### DISCUSSION TITLE
A/B Testing

### DISCUSSION THREAD

**Discussion by monsieurberry** [PiranhaCMS/piranha.core#1750] [Ideas]
Hi,

Wondering if A/B testing would ever be possible within the CMS? Either natively or via a plugin?

It would be great to be able to put up two different versions of a page and view result data on each one.

Appreciate it's a massive piece of work but just wanted to see if anyone had any thoughts on it or if it's ever been considered?

Thanks!
Adam

**Comment by tidyui** [comment]
It’s not something that we’ve considered, mostly because it’s a bit beyond the scope of what we’re trying to solve with Piranha.

If you’d want to give a try yourself I think maybe one unpublished page below the current page with the same slug could be used and then either custom controller or middleware code that selects one of the pages.

However this is of course just the tip of the iceberg as you will need to handle user state and then of course track conversion rates between the pages to get something out of it.

All in all it’s quite advanced functionality and nothing we’re planning on adding into the framework.

Best regards
